main/main.go

```diff
@@ -71,8 +71,6 @@ func main() {
 		Names:    map[string]uuid.UUID{},
 	}
 	p.Signatures = map[uuid.UUID][]byte{}
-	p.Certificates = map[string][]byte{}
```

Comment:
don't we need the Certificates map and the CSR map anymore?

Reply:
we don't store the Certificates and CSRs in the persistent storage anymore, but instead only create and send a certificate if the public key is not registered at the key service yet. The CSR will be generated once after reboot for every device (UUID). 
The problem here was that the protocol context file does not change if the user changes the backend environment and therefore the keys were not registered at the new stage